packages/calling/calling/src/common/Utils.ts

```diff
@@ -289,6 +289,176 @@ export function emitFinalFailure(emitterCb: LineErrorEmitterCallback, loggerCont
+            const caller = loggerContext.method || 'handleErrors';
+            await restoreRegCb(errorBody, caller);
+          }
+          break;
+        }
+        case DEVICE_ERROR_CODE.DEVICE_CREATION_DISABLED: {
+          const errorMessage =
+            'User is not configured for WebRTC calling. Please contact the administrator to resolve this issue.';
+          finalError = true;
+          updateLineErrorContext(
+            loggerContext,
+            ERROR_TYPE.FORBIDDEN_ERROR,
+            errorMessage,
+            LineStatus.INACTIVE,
+            lineError
+          );
+          log.warn(errorMessage, loggerContext);
+          emitterCb(lineError, true);
+          break;
+        }
+        case DEVICE_ERROR_CODE.DEVICE_CREATION_FAILED: {
+          const errorMessage =
+            'An unknown error occurred while provisioning the device. Wait a moment and try again.';
+          updateLineErrorContext(
+            loggerContext,
+            ERROR_TYPE.FORBIDDEN_ERROR,
+            errorMessage,
+            LineStatus.INACTIVE,
+            lineError
+          );
+          log.warn(errorMessage, loggerContext);
+          emitterCb(lineError, finalError);
+          break;
+        }
+        default: {
+          const errorMessage =
+            'An unknown error occurred. Wait a moment and try again. Please contact the administrator if the problem persists.';
+          updateLineErrorContext(
+            loggerContext,
+            ERROR_TYPE.FORBIDDEN_ERROR,
+            errorMessage,
+            LineStatus.INACTIVE,
+            lineError
+          );
+          log.warn(errorMessage, loggerContext);
+          emitterCb(lineError, finalError);
+        }
+      }
+      break;
+    }
+    case ERROR_CODE.DEVICE_NOT_FOUND: {
+      finalError = true;
+      log.warn(`404 Device Not Found`, loggerContext);
+
+      updateLineErrorContext(
+        loggerContext,
+        ERROR_TYPE.NOT_FOUND,
+        'The client has unregistered. Please wait for the client to register before attempting the call. If error persists, sign out, sign back in and attempt the call.',
+        LineStatus.INACTIVE,
+        lineError
+      );
+      emitterCb(lineError, finalError);
+      break;
+    }
+
+    default: {
+      updateLineErrorContext(
+        loggerContext,
+        ERROR_TYPE.DEFAULT,
+        'Unknown error',
+        LineStatus.INACTIVE,
+        lineError
+      );
+      log.warn(`Unknown Error`, loggerContext);
+      emitterCb(lineError, finalError);
+    }
+  }
+
+  return finalError;
+}
+
+/**
+ * Handle various Error flows related to calling client here. Decide whether to emit event or retry.
+ * @param err - Error body.
+ * @param emitterCb - CallingClientErrorEmitter
+ * @param loggerContext - Logging context that has method and file name
+ * @param restoreRegCb - Callback which will try restoring resgistration in case of 403
+ */
+export async function handleCallingClientErrors(
```

Comment:
It appears that the method `handleCallingClientErrors` is same as `handleRegistrationErrors`. 
If all of it same, why are we having two methods for same logic?
If not, Can we extract common logic into a private method?

Reply:
Removed unwanted ERROR codes from callingClient error handler after discussing with Priya.